app_test.go

```diff
@@ -0,0 +1,43 @@
+package main
+
+import (
+	"regexp"
+	"testing"
+
+	"github.com/Financial-Times/kafka-client-go/kafka"
+	hooks "github.com/Sirupsen/logrus/hooks/test"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHandleMessageInvalidWhitelist(t *testing.T) {
+	whitelist = regexp.MustCompile("http://cmdb.ft.com/systems/methode-web-pub")
```

Comment:
Again, the period characters should be escaped. With this value I can send a message with Origin-System-Id `http://cmdbAftBcom/systems/methode-web-pub` and it's passed by the whitelist.